mkosi/sandbox.py

```diff
@@ -98,8 +99,19 @@ def oserror(filename: str = "") -> None:
     raise OSError(ctypes.get_errno(), os.strerror(ctypes.get_errno()), filename or None)
 
 
+UNSHARE_EPERM_MSG = """
+mkosi was forbidden to unshare the user namespace. This probably means your distro has
+restricted unprivileged user namespace creation to reduce exposure to kernel bugs.
+For information about Ubuntu and AppArmor implementations of this restriction,
+see https://ubuntu.com/blog/ubuntu-23-10-restricted-unprivileged-user-namespaces.
+For other systems, research the kernel.unprivileged_userns_clone and
+user.max_user_namespaces sysctls.
+"""
+
 def unshare(flags: int) -> None:
     if libc.unshare(flags) < 0:
+        if ctypes.get_errno() == errno.EPERM:
```

Comment:
Oh only do this if unsharing userns. otherwise could lead to yet more pain & confusion

Reply:
Oh, interesting. It's not actually the `CLONE_NEWUSER` that fails. We're allowed to create userns but then inside that userns you don't get to create the actually interesting namespaces like the mount ns. 

I'm not gonna spend time digging into the exact details of this works since a) it'll be very Ubuntu-specific and b) it won't have much bearing on the ultimate action taken by the user. I'm just gonna be vaguer in the error message.